src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs

```diff
@@ -1312,5 +1312,166 @@ public void Match_Wordchar()
             }
             Assert.Empty(ambiguous);
         }
+
+        [Theory]
+        [InlineData(RegexOptions.None)]
+        [InlineData(RegexOptions.Compiled)]
+        [InlineData(RegexHelpers.RegexOptionNonBacktracking)]
+        public void StressTestDeepNestingOfConcat(RegexOptions options)
+        {
+            if (PlatformDetection.IsNetFramework && options == RegexHelpers.RegexOptionNonBacktracking)
+            {
+                return;
+            }
+
+            string pattern = string.Concat(Enumerable.Repeat("([a-z]", 1000).Concat(Enumerable.Repeat(")", 1000)));
+            string input = string.Concat(Enumerable.Repeat("abcde", 200));
+            var re = new Regex(pattern, options);
+            Assert.True(re.IsMatch(input));
+        }
+
+        public static IEnumerable<object[]> AllMatches_TestData()
+        {
+            foreach (RegexOptions options in new[] { RegexHelpers.RegexOptionNonBacktracking, RegexOptions.None, RegexOptions.Compiled })
+            {
+                if (PlatformDetection.IsNetFramework && options == RegexHelpers.RegexOptionNonBacktracking)
+                {
+                    yield break;
```

Comment:
Did you really mean `yield break;` here?  Shouldn't this be:
```suggestion
                    continue;
```
?

Reply:
I'm updating the `Regex.Match.Tests` to use the new `RegexHelpers.RegexOptionsExtended()` method instead to clean this up, whenever possible. The intent here was simply that the tests must not run in the `NonBacktracking` mode when `IsNetFramework` is true.

Reply:
> The intent here was simply that the tests must not run in the NonBacktracking mode when IsNetFramework is true.

Right, but `yield break;` will permanently exit the enumeration.

Reply:
This is not what I intended, should be fixed now with the commit I'm pushing next.